### DISCUSSION TITLE
Rust settings

### DISCUSSION THREAD

**Discussion by skinner12** [LazyVim/LazyVim#2874] [General]
Hello,
I'm studying rust and using lazyvim. I found that auto-completion and auto-import with this language is not working with my setup. I activated rust from lazyextras but it's not working like others programs languages.

For example if I want to open a file I start writing

`fs::`

and I'd like to get on popup options like `open` and so on. 

I'd like to have auto-import on top of file when I import crates like:

`use std::fs`.

I saw that formatting is working well.
`rust-analyzer` works too.

Can you help me to figure out this problems?

Thanks

**Comment by dpetka2001** [comment]
Do you use `rust-analyzer` from Mason or from your Rust toolchain?

**Reply by skinner12** [reply]
rust-analyzer shows as installed in mason

**Reply by dpetka2001** [reply]
Put the following spec in a file under `~/.config/nvim/lua/plugins/` directory
```lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        rust_analyzer = {
          mason = false,
        },
      },
    },
  },
}
```
Then uninstall `rust-analyzer` from Mason and make sure to add `rust-analyzer` component in your Rust toolchain.

**Reply by skinner12** [reply]
Thanks @dpetka2001, it works now! Thanks again for your support!

**Reply by nightfury99** [reply]
> Put the following spec in a file under `~/.config/nvim/lua/plugins/` directory
> 
> ```lua
> return {
>   {
>     "neovim/nvim-lspconfig",
>     opts = {
>       servers = {
>         rust_analyzer = {
>           mason = false,
>         },
>       },
>     },
>   },
> }
> ```
> 
> Then uninstall `rust-analyzer` from Mason and make sure to add `rust-analyzer` component in your Rust toolchain.

thank you veryy mucchh.
